code/modules/mob/living/carbon/carbon.dm

```diff
@@ -251,17 +251,22 @@
 					"<span class='notice'>You pat [src] on the head.</span>",\
 					)
 				else
-
-					M.visible_message(\
-					"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
-					"<span class='notice'>You hug [src].</span>",\
-					)
-					if(ishuman(src))
-						var/mob/living/carbon/human/H = src
-						if(H.wear_suit)
-							H.wear_suit.add_fingerprint(M)
-						else if(H.w_uniform)
-							H.w_uniform.add_fingerprint(M)
+					if(M.zone_selected == ("r_arm" || "r_hand" || "l_arm" || "l_hand"))
+						M.visible_message(\
+						"<span class='notice'>[M] shake [src] hand.</span>",\
+						"<span class='notice'>You shake hands with [src].</span>",\
+						)
```

Comment:
Should be `shakes` rather than `shake`, and `[src]'s` is better gramatically than just `[src]` for this:
```suggestion
						M.visible_message(\
						"<span class='notice'>[M] shakes [src]'s hand.</span>",\
						"<span class='notice'>You shake hands with [src].</span>",\
						)
```